### PR TITLE
ENH: clearer error message if read_hdf loads upsupported HDF file (GH9539)

### LINKED ISSUE #9539: Unclear error when loading unsupported HDF5 file
Backstory - I created some H5 files with h5py containing large 3D arrays (not python dataframes). Then I forgot this and tried to load with Pandas. Since they didn't contain real dataframes the load understandably  failed. However the error thrown was:

`TypeError: cannot create a storer if the object is not existing nor a value are passed`

(line 1148 in pandas.io.pytables) which apart from being awkward grammar (rather `cannot create a storer if the object does not exist nor a value given`) doesn't suggest anything about the h5 file not being supported.

Since passing around h5 files is fairly common and sometimes they contain tables and sometimes not, it would be nice if Pandas warned you clearly that the object you are trying to load isn't a pandas-compatible table.

### PR DESCRIPTION
**Pull request by a-y-khan**
- [x] closes #9539
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

The error message changed since the original bug report but still could be made more clear.

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -387,7 +387,10 @@ def read_hdf(
         if key is None:
             groups = store.groups()
             if len(groups) == 0:
-                raise ValueError("No dataset in HDF5 file.")
+                raise ValueError(
+                    "Dataset that is either compatible with Pandas data types "
```
Comment (jreback) [inline pandas/io/pytables.py:391]: incompatible

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -387,7 +387,10 @@ def read_hdf(
...
+                    "or is a table could not be found in HDF5 file."
+                )
```
Comment (jreback) [inline pandas/io/pytables.py:393]: can u add a file with an incompatible dataset (eg create in h5py)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/pytables.py (hunk 1 above)
Comment (a-y-khan) [reply pandas/io/pytables.py]: Updated. Is that better?

File: pandas/io/pytables.py (hunk 2 above)
Comment (a-y-khan) [reply pandas/io/pytables.py:393]: Working on it...

Comment (a-y-khan) [reply pandas/io/pytables.py:393]: OK, added a file with an incompatible `dataset, pandas/tests/io/data/legacy_hdf/incompatible_dataset.h5` and a test

File: doc/source/whatsnew/v1.1.0.rst [hunk 3]
```diff
@@ -98,6 +98,8 @@ Other enhancements
   This can be used to set a custom compression level, e.g.,
   ``df.to_csv(path, compression={'method': 'gzip', 'compresslevel': 1}``
   (:issue:`33196`)
+- :func:`pandas.read_hdf` has a more explicit error message when loading an
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:101]: can you move to the IO section

**Review by jreback — CHANGES_REQUESTED** [review]
this looks good, if you can just move the whatsnew note and ping on green for merge.

**Review by jreback — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @a-y-khan

**Pull request merged**